packages/programs/mud.config.ts

```diff
@@ -0,0 +1,54 @@
+import { defineWorld } from "@latticexyz/world";
+
+export default defineWorld({
+  // Note: this is required as the world is deployed with this
+  deploy: {
+    upgradeableWorldImplementation: true,
+  },
+  namespace: "default-1",
+  userTypes: {
+    ObjectTypeId: {
+      filePath: "@dust/world/src/ObjectTypeId.sol",
+      type: "uint16",
+    },
+    EntityId: { filePath: "@dust/world/src/EntityId.sol", type: "bytes32" },
+    ResourceId: {
+      filePath: "@latticexyz/store/src/ResourceId.sol",
+      type: "bytes32",
+    },
+    ProgramId: {
+      filePath: "@dust/world/src/ProgramId.sol",
+      type: "bytes32",
+    },
+  },
+  tables: {
+    Admin: {
+      schema: {
+        entityId: "EntityId",
+        admin: "address",
+      },
+      key: ["entityId"],
+    },
+    AllowedPlayers: {
+      schema: {
+        entityId: "EntityId",
+        players: "address[]",
```

Comment:
wdyt of having `(smart entity, caller) => bool allowed` plus a counter? then you can set allowed entities individually vs having to send the whole array

Reply:
one problem is how to do cleanup?

`onDetachProgram`, we wouldnt be able to clean up the old allowed values. If someone then places a new smart item in the same position, they'd get the same entity id (since its tied to position), and it would incorrectly have the previous approved callers

Reply:
with the counter you can iterate through all of them, or you can do it as in the shop program where there is an indirection chest => shopId => shop info.

shopId is set incrementally on attachment, so it is always unique and you just use a new shop